src/main/src/test/java/org/geoserver/catalog/retype/MockRetypedFeatureConverter.java

```diff
@@ -0,0 +1,142 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog.retype;
+
+import static java.util.Optional.ofNullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.MockRetypeFeatureTypeCallback;
+import org.geotools.data.Query;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Point;
+import org.opengis.feature.Property;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+
+/*
+ * A Mock Converter Helper class to demonstrate how to
+ * Retype Feature by generating Geometries on the fly
+ * Defining Geomtry attribute on the fly
+ * Query handling which ensure that dynamic field is not delegated to Data Store
+ *
+ * **/
+public class MockRetypedFeatureConverter {

```

Comment:
Why this generic ``Mock```names? This doesn't mock anything and its not generic no? This implements a concrete behavior to validate an extension point no? Why not put this class inside the test a private static inner classes?

Reply:
will change the name, its being used in two classes as a helper class hence cannot be made private inner static